### ISSUE TITLE
setup-repositories fails on macOS

### ISSUE THREAD

**Issue by rickmanelius** [cerc-io/stack-orchestrator#77]
I verified I have a working PAT and can run a similar clone of another repo using `git clone https://github.com/rickmanelius/my-repo.git`. However, when I run `laconic-so --verbose setup-repositories`, I keep getting blocked by what seems to be 

https://gist.github.com/rickmanelius/6eb52c5a66aedaaf76bfd8a815b5d84b

I also reviewed the standard solutions here, but came up with nothing https://stackoverflow.com/questions/9617336/how-to-resolve-git-did-not-exit-cleanly-exit-cod

**Comment by dboreham** [comment]
Whoa. That repo shouldn't be cloned in stack-orchestrator. Checking why we have that in the list...

**Comment by dboreham** [comment]
Please try again (git pull updated stack-orchestrator files): I removed the failed repo.

Also recommend using `laconic-so setup-repositories --git-ssh` and `ssh-agent` to avoid re-entering git credentials.

**Comment by rickmanelius** [comment]
Hi @dboreham. Thanks for this. I have two different results to report:

* If I don't use `venv`, this still fails at `vulcanize/ops`. I assume because I need to generate a new binary for `laconic-so`
* If I do use `venv`, I get all the repos until `ipld-eth-beacon-indexer`. I assume this is because it's private. I may have to update my PAT to get around this.

**Comment by zramsay** [comment]
fresh install of Docker on MacOS throws this error on `build-repositories`:

 `error storing credentials - err: exec: "docker-credential-desktop": executable file not found in $PATH, out:` 
 might be because i didn't completely uninstall an old, broken version of docker for mac
 
 this fix worked:
 https://github.com/docker/for-mac/issues/3774#issuecomment-518600892

**Comment by dboreham** [comment]
> this fix worked: [docker/for-mac#3774 (comment)](https://github.com/docker/for-mac/issues/3774#issuecomment-518600892)

Oh no! Dockerco *still* haven't fixed that bug!!

**Comment by rickmanelius** [comment]
This is now resolved:
* Private Repos removed from a branch about to be committed https://github.com/deep-stack/stack-orchestrator/tree/pm-erc20-watcher-stack.
* Use of `venv` to get around needing a new binary release

I'll leave this ticket open and repurpose to a general macOS setup ticket. Also happy to close and spin up a new ticket should another issue arise.

**Comment by rickmanelius** [comment]
@zramsay I ran into that same snag. Given how pervasive this is, do we need to note that in the requirements for Docker in the README.md? Notice... you need to change a character in the configuration file here https://forums.docker.com/t/docker-credential-desktop-exe-executable-file-not-found-in-path-using-wsl2/100225/5

**Comment by dboreham** [comment]
We shouldn't be trying to support docker's product and in general it would be wrong to tell users to edit their docker config file.

A page of common issues with tooling would be good though, and a link to that from our readme.

**Comment by zramsay** [comment]
what @dboreham said

**Comment by rickmanelius** [comment]
I'm good with "common issues" or "troubleshooting" section. 

I'll consider this issue closed then. Others hitting the snag can find in this issue queue, and if it's pervasive, we can re-open and get a PR open for troubleshooting.
